### PR TITLE
Codgen updates for linter

### PR DESCRIPTION
**Pull request by TimWhiting**
Here are the updates for the codegen

### CODE REVIEW

File: packages/riverpod_lint/bin/custom_lint.dart [hunk 1]
```diff
@@ -16,7 +16,7 @@ import 'package:riverpod_lint/src/type_checker.dart';
 const _providerBase =
     TypeChecker.fromName('ProviderBase', packageName: 'riverpod');
 const _autoDispose =
-    TypeChecker.fromName('AutoDisposeProviderBase', packageName: 'riverpod');
+    TypeChecker.fromName('AutoDisposeProvider', packageName: 'riverpod');
```
[the remark this answers is not in the document]
Comment (rrousselGit) [reply packages/riverpod_lint/bin/custom_lint.dart:19]: The logic for supporting the removal of `AutoDisposeProviderBase` would probably be to swap the checks

Insead we can check whether a provider is an instance of `AlwaysAliveProviderBase`. And if it's not, then it's an autoDispose

[the remark this answers is not in the document]
Comment (TimWhiting) [reply packages/riverpod_lint/bin/custom_lint.dart]: Done, but alternatively, we could have done TypeChecker.any with AutoDisposeProviderBase + AutoDiposeProvider

[the remark this answers is not in the document]
Comment (rrousselGit) [reply packages/riverpod_lint/bin/custom_lint.dart]: That would have the same provider. We still can't use AutoDisposeProvider

File: packages/riverpod_lint/bin/custom_lint.dart [hunk 2]
```diff
@@ -15,8 +15,14 @@ import 'package:riverpod_lint/src/type_checker.dart';
 
 const _providerBase =
     TypeChecker.fromName('ProviderBase', packageName: 'riverpod');
-const _autoDispose =
-    TypeChecker.fromName('AutoDisposeProviderBase', packageName: 'riverpod');
+const _nonAutoDispose = TypeChecker.any([
+  TypeChecker.fromName('AlwaysAliveProviderBase', packageName: 'riverpod'),
```
Comment (rrousselGit) [inline packages/riverpod_lint/bin/custom_lint.dart:19]: This line is redundant, considering `AlwaysAliveProviderBase` is an `AlwaysAliveProviderListenable`

File: packages/riverpod_lint/bin/custom_lint.dart [hunk 3]
```diff
@@ -15,8 +15,14 @@ import 'package:riverpod_lint/src/type_checker.dart';
 
 const _providerBase =
     TypeChecker.fromName('ProviderBase', packageName: 'riverpod');
-const _autoDispose =
-    TypeChecker.fromName('AutoDisposeProviderBase', packageName: 'riverpod');
+const _nonAutoDispose = TypeChecker.any([
```
Comment (rrousselGit) [inline packages/riverpod_lint/bin/custom_lint.dart:18]: Could you rename it to `_alwaysAliveProviderListenable`?

That'd avoid the double negation of `(!_nonAutoDispose)`

File: packages/riverpod_lint/bin/custom_lint.dart [hunk 4]
```diff
@@ -813,6 +850,11 @@ Then dispose of the listener when you no longer need the autoDispose provider to
   }
 
   Stream<Lint> _checkValidProviderDeclaration(AstNode providerNode) async* {
+    if (providerNode is ClassDeclaration ||
+        providerNode is FunctionDeclaration) {
+      // Codegen provider
```
Comment (rrousselGit) [inline packages/riverpod_lint/bin/custom_lint.dart:855]: Unrelated but one thing the linter could do is expose quick fixes, to add things such as:
- automatically adding the "extends _$Foo"
- add the missing `build` method
- when using a function, fix any typo in the Ref type (so no `@riverpod foo(BarRef ref)`)

**Review by rrousselGit — APPROVED** [review]
LGTM with a few nits

File: packages/riverpod_lint/bin/custom_lint.dart (hunk 4 above)
Comment (rrousselGit) [reply packages/riverpod_lint/bin/custom_lint.dart:855]: No pressure. :smile:

**Comment by rrousselGit** [comment]
Thanks!

**Pull request merged**